File: navigation/src/main/java/com/example/navigation/stepsEngine/field/ServiceCodeField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class ServiceCodeField extends Field {

    @Override
    public String get(FlowState flowState) {
        return flowState.getCard().getServiceCode();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/ServiceCodeField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class ServiceCodeField : Field() {
    override operator fun get(flowState: FlowState): String? {
        return flowState.card.serviceCode
    }

    override fun getId(): String {
        return FieldName.SERVICE_CODE
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}